### PR TITLE
Fixed example_hear_grap.py

### PR DESCRIPTION
**Pull request by shanki98**
Updated example to not use ears.recognize. If ears part is not used anymore, can we remove it from Master?

This example is additionally not complete because it does not search for the object.

### CODE REVIEW

File: robot_smach_states/examples/example_hear_grasp.py [hunk 1]
```diff
@@ -13,13 +13,13 @@ def __init__(self, robot):
...
-                                           choices={'drink': ['coke', 'fanta']},
-                                           time_out=rospy.Duration(60))
-        choices = result.choices['drink'] if result else None
-        if choices:
-            rospy.loginfo('I heared:' + repr(choices))
-        userdata.heared = choices
+        answer = self.robot.hmi.query(description='can i have a coke?',
+                                      grammar="T -> DET coke \n DET -> a",
+                                      target='T',
+                                      timeout=rospy.Duration(60).to_sec())
```
Comment (MatthijsBurgh) [inline robot_smach_states/examples/example_hear_grasp.py:19]: Going from a number to a `rospy.Duration` to a number......

File: robot_smach_states/examples/example_hear_grasp.py [hunk 2]
```diff
@@ -13,13 +13,13 @@ def __init__(self, robot):
...
 
     def execute(self, userdata):
-        result = self.robot.ears.recognize(spec='Bring me a <drink>',
-                                           choices={'drink': ['coke', 'fanta']},
-                                           time_out=rospy.Duration(60))
-        choices = result.choices['drink'] if result else None
-        if choices:
-            rospy.loginfo('I heared:' + repr(choices))
-        userdata.heared = choices
+        answer = self.robot.hmi.query(description='can i have a coke?',
```
Comment (MatthijsBurgh) [inline robot_smach_states/examples/example_hear_grasp.py:16]: Now we can only hear coke....

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
The examples should be a kind of a test, but this isn't really testing any smach state. It is only using the initialize. I would suggest to remove it.

File: robot_smach_states/examples/example_hear_grasp.py (hunk 2 above)
Comment (shanki98) [reply robot_smach_states/examples/example_hear_grasp.py:16]: This is an example for query I suppose, so we are not hearing anything from the Robot.

Comment (MatthijsBurgh) [reply robot_smach_states/examples/example_hear_grasp.py:16]: Please restore old coke/fanta option

File: robot_smach_states/examples/example_hear_grasp.py [hunk 3]
```diff
@@ -35,9 +35,10 @@ def execute(self, userdata):
 
         rospy.loginfo('I\'m going to search for {}'.format(type))
         # call robot.ed.get_closest_entity()
-        entity = {'id': 'coke-1'}
+        entity = self.robot.ed.get_closest_entity(etype=userdata.heard)
+        #entity = {'id': 'coke-1'}
```
Comment (MatthijsBurgh) [inline robot_smach_states/examples/example_hear_grasp.py:39]: Remove commented code

File: robot_smach_states/examples/example_hear_grasp.py [hunk 4]
```diff
@@ -35,9 +35,10 @@ def execute(self, userdata):
...
 
-        userdata.entity = entity
+        userdata.entity = entsity
```
Comment (MatthijsBurgh) [inline robot_smach_states/examples/example_hear_grasp.py:41]: Typo

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

**Comment by shanki98** [comment]
The example is now fixed to hear both coke and fanta. But the description is just a query and not what the robot hears. If we want the example to hear a sentence then we must use speech recognition as well in this example and this would complicate the example further. SO this example tests 
1. hmi.query
2. get_closest_entity 
3. Grammar parser

File: robot_smach_states/examples/example_hear_grasp.py [hunk 5]
```diff
@@ -9,33 +9,32 @@
...
-        choices = result.choices['drink'] if result else None
-        if choices:
-            rospy.loginfo('I heared:' + repr(choices))
-        userdata.heared = choices
+        answer = self.robot.hmi.query(description='can i have a fanta?',
+                                      grammar="T -> DET coke | DET fanta \n DET -> a | an ",
+                                      target='T',
+                                      timeout=60)
+        if answer:
+            rospy.loginfo('I heard:' + answer.sentence)
```
Comment (PetervDooren) [inline robot_smach_states/examples/example_hear_grasp.py:21]: As this is an example: meant to showcase the skill. I would make the robot say this out loud. Now you can only see this in a terminal.

File: robot_smach_states/examples/example_hear_grasp.py [hunk 6]
```diff
@@ -9,33 +9,32 @@
...
-        type = userdata.type
+        type = userdata.heard
 
         rospy.loginfo('I\'m going to search for {}'.format(type))
-        # call robot.ed.get_closest_entity()
-        entity = {'id': 'coke-1'}
+        entity = self.robot.ed.get_closest_entity(etype=userdata.heard)
 
         userdata.entity = entity
         return 'succeeded'
```
Comment (PetervDooren) [inline robot_smach_states/examples/example_hear_grasp.py:40]: As Matthijs also mentioned, this information is not used. But that was already an issue with this example. I don't think you need to fix this right now.

**Review by PetervDooren — DISMISSED** [review]
Some suggestions for improvement, But this is alright

**Comment by shanki98** [comment]
@MatthijsBurgh can this example be merged?

**Review by MatthijsBurgh — DISMISSED** [review]
Merge this damn thing.

**Pull request closed without merging**